File: cs50/pset6/cash/cash.py
# Calculate minimum number of coins required to give a user change

from cs50 import get_float

# Reprompt user until input is greater than zero
while True:
    change = get_float("Change owed: ") * 100
    if change > 0:
        break

quarters = dimes = nickels = pennies = 0

if change >= 25:
    quarters = int(change / 25)
    change = change % 25
if change >= 10:
    dimes = int(change / 10)
    change = change % 10
if change >= 5:
    nickels = int(change / 5)
    change = change % 5
if change >= 1:
    pennies = int(change / 1)
    change = change % 1

# Add up number of coins and print total
total = quarters + dimes + nickels + pennies

print(total)

File: cs50/pset1/cash/cash.c
// Problem Set 1: Change in Coins

#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Prompt for change owed
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // Convert change to cents
    int cents = round(change * 100);

    // Calculate coins
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;

    while (cents > 0)
    {
        while (cents >= 5)
        {
            while (cents >= 10)
            {
                while (cents >= 25)
                {
                    quarters = cents / 25;
                    cents = cents % 25;
                }

                dimes = cents / 10;
                cents = cents % 10;
            }

            nickels = cents / 5;
            cents = cents % 5;
        }

        pennies = cents / 1;
        cents = cents % 1;
    }

    // Add and print total number of coins
    int coins = quarters + dimes + nickels + pennies;
    printf("%i\n", coins);
}